core/src/main/java/org/apache/cxf/endpoint/ClientImpl.java

```diff
@@ -658,7 +659,7 @@ private void enrichFault(Fault fault) {
             throw ex;
         }
 
-        if (resList == null   
+        if (Boolean.getBoolean(CHECK_EMTPY_RESPONSE) && resList == null   
```

Comment:
it goes through a Properties access so it is synchronized, maybe read it once per client instance to avoid an useless locking?